doc/langref.html.in

```diff
@@ -5458,9 +5458,46 @@ test "@intToPtr for pointer to zero bit type" {
       {#header_close#}
 
       {#header_open|Result Location Semantics#}
-      <p>
-      <a href="https://github.com/ziglang/zig/issues/2809">TODO add documentation for this</a>
-      </p>
+      <p>Certain expressions which manipulate a value in memory will directly initialize the final location of
+      that memory, to avoid making multiple allocations and copying between them:</p>
+      {#code_begin#}
+pub fn main() void {
+    // foo() allocates its result directly in x -- only one allocation is made
```

Comment:
"Allocation" is usually considered dynamic allocation, but that's not what's happening here.
Here we're talking about more general allocation of registers or stack space for variables.
For `field2` below, not even that - the allocation of space was already done by `var y` so we're just referring to that space instead.

Reply:
Ok, what would the appropriate term for that be then?

Reply:
I'm not sure. I guess you could just clarify it; that the term refers to allocation of variables, which is either on the stack or in a register, rather than a dynamic allocation. 🤔

Reply:
How about just qualify the allocation as a stack allocation?

Reply:
Because it may (and probably won't) always be a stack allocation, and you might care - Zig being a systems language - about that fact.

OTOH, you could argue that putting it in a register is an optimization, and that it's easier to think about if you just talk about the stack -- even though it's an optimization that always happens.

Reply:
Ok, then 'non-heap' allocation, if you want to factor in compiler optimization potentials.

Reply:
Ok, cleared up and rebased. Now it should be ready for merge.

Reply:
Also subscribed to #2761 and #2765 so I can fix those points when it's relevant (haven't yet gotten the hang of inline testing).